// devextreme-angular/esm5/ui/nested/scale-type-range.js
/*!
 * devextreme-angular
 * Version: 21.2.15
 * Build date: Tue Dec 12 2023
 *
 * Copyright (c) 2012 - 2023 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
import * as tslib_1 from "tslib";
/* tslint:disable:max-line-length */
import { Component, NgModule, Host, SkipSelf, Input } from '@angular/core';
import { NestedOptionHost, } from 'devextreme-angular/core';
import { NestedOption } from 'devextreme-angular/core';
var DxoScaleTypeRangeComponent = /** @class */ (function (_super) {
    tslib_1.__extends(DxoScaleTypeRangeComponent, _super);
    function DxoScaleTypeRangeComponent(parentOptionHost, optionHost) {
        var _this = _super.call(this) || this;
        parentOptionHost.setNestedOption(_this);
        optionHost.setHost(_this, _this._fullOptionPath.bind(_this));
        return _this;
    }
    Object.defineProperty(DxoScaleTypeRangeComponent.prototype, "max", {
        get: function () {
            return this._getOption('max');
        },
        set: function (value) {
            this._setOption('max', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoScaleTypeRangeComponent.prototype, "min", {
        get: function () {
            return this._getOption('min');
        },
        set: function (value) {
            this._setOption('min', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoScaleTypeRangeComponent.prototype, "_optionPath", {
        get: function () {
            return 'scaleTypeRange';
        },
        enumerable: true,
        configurable: true
    });
    DxoScaleTypeRangeComponent.prototype.ngOnInit = function () {
        this._addRecreatedComponent();
    };
    DxoScaleTypeRangeComponent.prototype.ngOnDestroy = function () {
        this._addRemovedOption(this._getOptionPath());
    };
    DxoScaleTypeRangeComponent.ctorParameters = function () { return [
        { type: NestedOptionHost, decorators: [{ type: SkipSelf }, { type: Host }] },
        { type: NestedOptionHost, decorators: [{ type: Host }] }
    ]; };
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], DxoScaleTypeRangeComponent.prototype, "max", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], DxoScaleTypeRangeComponent.prototype, "min", null);
    DxoScaleTypeRangeComponent = tslib_1.__decorate([
        Component({
            selector: 'dxo-scale-type-range',
            template: '',
            providers: [NestedOptionHost],
            styles: ['']
        }),
        tslib_1.__param(0, SkipSelf()), tslib_1.__param(0, Host()),
        tslib_1.__param(1, Host()),
        tslib_1.__metadata("design:paramtypes", [NestedOptionHost,
            NestedOptionHost])
    ], DxoScaleTypeRangeComponent);
    return DxoScaleTypeRangeComponent;
}(NestedOption));
export { DxoScaleTypeRangeComponent };
var DxoScaleTypeRangeModule = /** @class */ (function () {
    function DxoScaleTypeRangeModule() {
    }
    DxoScaleTypeRangeModule = tslib_1.__decorate([
        NgModule({
            declarations: [
                DxoScaleTypeRangeComponent
            ],
            exports: [
                DxoScaleTypeRangeComponent
            ],
        })
    ], DxoScaleTypeRangeModule);
    return DxoScaleTypeRangeModule;
}());
export { DxoScaleTypeRangeModule };
