// devextreme-angular/bundles/devextreme-angular.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('devextreme-angular/core'), require('@angular/core'), require('devextreme-angular/ui/accordion'), require('devextreme-angular/ui/action-sheet'), require('devextreme-angular/ui/autocomplete'), require('devextreme-angular/ui/bar-gauge'), require('devextreme-angular/ui/box'), require('devextreme-angular/ui/bullet'), require('devextreme-angular/ui/button'), require('devextreme-angular/ui/button-group'), require('devextreme-angular/ui/calendar'), require('devextreme-angular/ui/chart'), require('devextreme-angular/ui/check-box'), require('devextreme-angular/ui/circular-gauge'), require('devextreme-angular/ui/color-box'), require('devextreme-angular/ui/context-menu'), require('devextreme-angular/ui/data-grid'), require('devextreme-angular/ui/date-box'), require('devextreme-angular/ui/defer-rendering'), require('devextreme-angular/ui/diagram'), require('devextreme-angular/ui/draggable'), require('devextreme-angular/ui/drawer'), require('devextreme-angular/ui/drop-down-box'), require('devextreme-angular/ui/drop-down-button'), require('devextreme-angular/ui/file-manager'), require('devextreme-angular/ui/file-uploader'), require('devextreme-angular/ui/filter-builder'), require('devextreme-angular/ui/form'), require('devextreme-angular/ui/funnel'), require('devextreme-angular/ui/gallery'), require('devextreme-angular/ui/gantt'), require('devextreme-angular/ui/html-editor'), require('devextreme-angular/ui/linear-gauge'), require('devextreme-angular/ui/list'), require('devextreme-angular/ui/load-indicator'), require('devextreme-angular/ui/load-panel'), require('devextreme-angular/ui/lookup'), require('devextreme-angular/ui/map'), require('devextreme-angular/ui/menu'), require('devextreme-angular/ui/multi-view'), require('devextreme-angular/ui/nav-bar'), require('devextreme-angular/ui/number-box'), require('devextreme-angular/ui/pie-chart'), require('devextreme-angular/ui/pivot-grid'), require('devextreme-angular/ui/pivot-grid-field-chooser'), require('devextreme-angular/ui/polar-chart'), require('devextreme-angular/ui/popover'), require('devextreme-angular/ui/popup'), require('devextreme-angular/ui/progress-bar'), require('devextreme-angular/ui/radio-group'), require('devextreme-angular/ui/range-selector'), require('devextreme-angular/ui/range-slider'), require('devextreme-angular/ui/recurrence-editor'), require('devextreme-angular/ui/resizable'), require('devextreme-angular/ui/responsive-box'), require('devextreme-angular/ui/sankey'), require('devextreme-angular/ui/scheduler'), require('devextreme-angular/ui/scroll-view'), require('devextreme-angular/ui/select-box'), require('devextreme-angular/ui/slide-out'), require('devextreme-angular/ui/slide-out-view'), require('devextreme-angular/ui/slider'), require('devextreme-angular/ui/sortable'), require('devextreme-angular/ui/sparkline'), require('devextreme-angular/ui/speed-dial-action'), require('devextreme-angular/ui/switch'), require('devextreme-angular/ui/tab-panel'), require('devextreme-angular/ui/tabs'), require('devextreme-angular/ui/tag-box'), require('devextreme-angular/ui/text-area'), require('devextreme-angular/ui/text-box'), require('devextreme-angular/ui/tile-view'), require('devextreme-angular/ui/toast'), require('devextreme-angular/ui/toolbar'), require('devextreme-angular/ui/tooltip'), require('devextreme-angular/ui/tree-list'), require('devextreme-angular/ui/tree-map'), require('devextreme-angular/ui/tree-view'), require('devextreme-angular/ui/validation-group'), require('devextreme-angular/ui/validation-summary'), require('devextreme-angular/ui/validator'), require('devextreme-angular/ui/vector-map')) :
    typeof define === 'function' && define.amd ? define('devextreme-angular', ['exports', 'devextreme-angular/core', '@angular/core', 'devextreme-angular/ui/accordion', 'devextreme-angular/ui/action-sheet', 'devextreme-angular/ui/autocomplete', 'devextreme-angular/ui/bar-gauge', 'devextreme-angular/ui/box', 'devextreme-angular/ui/bullet', 'devextreme-angular/ui/button', 'devextreme-angular/ui/button-group', 'devextreme-angular/ui/calendar', 'devextreme-angular/ui/chart', 'devextreme-angular/ui/check-box', 'devextreme-angular/ui/circular-gauge', 'devextreme-angular/ui/color-box', 'devextreme-angular/ui/context-menu', 'devextreme-angular/ui/data-grid', 'devextreme-angular/ui/date-box', 'devextreme-angular/ui/defer-rendering', 'devextreme-angular/ui/diagram', 'devextreme-angular/ui/draggable', 'devextreme-angular/ui/drawer', 'devextreme-angular/ui/drop-down-box', 'devextreme-angular/ui/drop-down-button', 'devextreme-angular/ui/file-manager', 'devextreme-angular/ui/file-uploader', 'devextreme-angular/ui/filter-builder', 'devextreme-angular/ui/form', 'devextreme-angular/ui/funnel', 'devextreme-angular/ui/gallery', 'devextreme-angular/ui/gantt', 'devextreme-angular/ui/html-editor', 'devextreme-angular/ui/linear-gauge', 'devextreme-angular/ui/list', 'devextreme-angular/ui/load-indicator', 'devextreme-angular/ui/load-panel', 'devextreme-angular/ui/lookup', 'devextreme-angular/ui/map', 'devextreme-angular/ui/menu', 'devextreme-angular/ui/multi-view', 'devextreme-angular/ui/nav-bar', 'devextreme-angular/ui/number-box', 'devextreme-angular/ui/pie-chart', 'devextreme-angular/ui/pivot-grid', 'devextreme-angular/ui/pivot-grid-field-chooser', 'devextreme-angular/ui/polar-chart', 'devextreme-angular/ui/popover', 'devextreme-angular/ui/popup', 'devextreme-angular/ui/progress-bar', 'devextreme-angular/ui/radio-group', 'devextreme-angular/ui/range-selector', 'devextreme-angular/ui/range-slider', 'devextreme-angular/ui/recurrence-editor', 'devextreme-angular/ui/resizable', 'devextreme-angular/ui/responsive-box', 'devextreme-angular/ui/sankey', 'devextreme-angular/ui/scheduler', 'devextreme-angular/ui/scroll-view', 'devextreme-angular/ui/select-box', 'devextreme-angular/ui/slide-out', 'devextreme-angular/ui/slide-out-view', 'devextreme-angular/ui/slider', 'devextreme-angular/ui/sortable', 'devextreme-angular/ui/sparkline', 'devextreme-angular/ui/speed-dial-action', 'devextreme-angular/ui/switch', 'devextreme-angular/ui/tab-panel', 'devextreme-angular/ui/tabs', 'devextreme-angular/ui/tag-box', 'devextreme-angular/ui/text-area', 'devextreme-angular/ui/text-box', 'devextreme-angular/ui/tile-view', 'devextreme-angular/ui/toast', 'devextreme-angular/ui/toolbar', 'devextreme-angular/ui/tooltip', 'devextreme-angular/ui/tree-list', 'devextreme-angular/ui/tree-map', 'devextreme-angular/ui/tree-view', 'devextreme-angular/ui/validation-group', 'devextreme-angular/ui/validation-summary', 'devextreme-angular/ui/validator', 'devextreme-angular/ui/vector-map'], factory) :
    (global = global || self, factory(global['devextreme-angular'] = {}, global['devextreme-angular'].core, global.ng.core, global['devextreme-angular'].ui.accordion, global['devextreme-angular'].ui['action-sheet'], global['devextreme-angular'].ui.autocomplete, global['devextreme-angular'].ui['bar-gauge'], global['devextreme-angular'].ui.box, global['devextreme-angular'].ui.bullet, global['devextreme-angular'].ui.button, global['devextreme-angular'].ui['button-group'], global['devextreme-angular'].ui.calendar, global['devextreme-angular'].ui.chart, global['devextreme-angular'].ui['check-box'], global['devextreme-angular'].ui['circular-gauge'], global['devextreme-angular'].ui['color-box'], global['devextreme-angular'].ui['context-menu'], global['devextreme-angular'].ui['data-grid'], global['devextreme-angular'].ui['date-box'], global['devextreme-angular'].ui['defer-rendering'], global['devextreme-angular'].ui.diagram, global['devextreme-angular'].ui.draggable, global['devextreme-angular'].ui.drawer, global['devextreme-angular'].ui['drop-down-box'], global['devextreme-angular'].ui['drop-down-button'], global['devextreme-angular'].ui['file-manager'], global['devextreme-angular'].ui['file-uploader'], global['devextreme-angular'].ui['filter-builder'], global['devextreme-angular'].ui.form, global['devextreme-angular'].ui.funnel, global['devextreme-angular'].ui.gallery, global['devextreme-angular'].ui.gantt, global['devextreme-angular'].ui['html-editor'], global['devextreme-angular'].ui['linear-gauge'], global['devextreme-angular'].ui.list, global['devextreme-angular'].ui['load-indicator'], global['devextreme-angular'].ui['load-panel'], global['devextreme-angular'].ui.lookup, global['devextreme-angular'].ui.map, global['devextreme-angular'].ui.menu, global['devextreme-angular'].ui['multi-view'], global['devextreme-angular'].ui['nav-bar'], global['devextreme-angular'].ui['number-box'], global['devextreme-angular'].ui['pie-chart'], global['devextreme-angular'].ui['pivot-grid'], global['devextreme-angular'].ui['pivot-grid-field-chooser'], global['devextreme-angular'].ui['polar-chart'], global['devextreme-angular'].ui.popover, global['devextreme-angular'].ui.popup, global['devextreme-angular'].ui['progress-bar'], global['devextreme-angular'].ui['radio-group'], global['devextreme-angular'].ui['range-selector'], global['devextreme-angular'].ui['range-slider'], global['devextreme-angular'].ui['recurrence-editor'], global['devextreme-angular'].ui.resizable, global['devextreme-angular'].ui['responsive-box'], global['devextreme-angular'].ui.sankey, global['devextreme-angular'].ui.scheduler, global['devextreme-angular'].ui['scroll-view'], global['devextreme-angular'].ui['select-box'], global['devextreme-angular'].ui['slide-out'], global['devextreme-angular'].ui['slide-out-view'], global['devextreme-angular'].ui.slider, global['devextreme-angular'].ui.sortable, global['devextreme-angular'].ui.sparkline, global['devextreme-angular'].ui['speed-dial-action'], global['devextreme-angular'].ui.switch, global['devextreme-angular'].ui['tab-panel'], global['devextreme-angular'].ui.tabs, global['devextreme-angular'].ui['tag-box'], global['devextreme-angular'].ui['text-area'], global['devextreme-angular'].ui['text-box'], global['devextreme-angular'].ui['tile-view'], global['devextreme-angular'].ui.toast, global['devextreme-angular'].ui.toolbar, global['devextreme-angular'].ui.tooltip, global['devextreme-angular'].ui['tree-list'], global['devextreme-angular'].ui['tree-map'], global['devextreme-angular'].ui['tree-view'], global['devextreme-angular'].ui['validation-group'], global['devextreme-angular'].ui['validation-summary'], global['devextreme-angular'].ui.validator, global['devextreme-angular'].ui['vector-map']));
}(this, (function (exports, core, core$1, accordion, actionSheet, autocomplete, barGauge, box, bullet, button, buttonGroup, calendar, chart, checkBox, circularGauge, colorBox, contextMenu, dataGrid, dateBox, deferRendering, diagram, draggable, drawer, dropDownBox, dropDownButton, fileManager, fileUploader, filterBuilder, form, funnel, gallery, gantt, htmlEditor, linearGauge, list, loadIndicator, loadPanel, lookup, map, menu, multiView, navBar, numberBox, pieChart, pivotGrid, pivotGridFieldChooser, polarChart, popover, popup, progressBar, radioGroup, rangeSelector, rangeSlider, recurrenceEditor, resizable, responsiveBox, sankey, scheduler, scrollView, selectBox, slideOut, slideOutView, slider, sortable, sparkline, speedDialAction, _switch, tabPanel, tabs, tagBox, textArea, textBox, tileView, toast, toolbar, tooltip, treeList, treeMap, treeView, validationGroup, validationSummary, validator, vectorMap) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __rest(s, e) {
        var t = {};
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __createBinding(o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
    }

    function __exportStar(m, exports) {
        for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) exports[p] = m[p];
    }

    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m) return m.call(o);
        if (o && typeof o.length === "number") return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    };

    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }

    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
    }

    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }

    function __asyncValues(o) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
    }

    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
        return cooked;
    };

    function __importStar(mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
        result.default = mod;
        return result;
    }

    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }

    function __classPrivateFieldGet(receiver, privateMap) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to get private field on non-instance");
        }
        return privateMap.get(receiver);
    }

    function __classPrivateFieldSet(receiver, privateMap, value) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to set private field on non-instance");
        }
        privateMap.set(receiver, value);
        return value;
    }

    /*!
     * devextreme-angular
     * Version: 21.2.15
     * Build date: Tue Dec 12 2023
     *
     * Copyright (c) 2012 - 2023 Developer Express Inc. ALL RIGHTS RESERVED
     *
     * This software may be modified and distributed under the terms
     * of the MIT license. See the LICENSE file in the root of the project for details.
     *
     * https://github.com/DevExpress/devextreme-angular
     */
    var DevExtremeModule = /** @class */ (function () {
        function DevExtremeModule() {
        }
        DevExtremeModule = __decorate([
            core$1.NgModule({
                imports: [
                    accordion.DxAccordionModule,
                    actionSheet.DxActionSheetModule,
                    autocomplete.DxAutocompleteModule,
                    barGauge.DxBarGaugeModule,
                    box.DxBoxModule,
                    bullet.DxBulletModule,
                    button.DxButtonModule,
                    buttonGroup.DxButtonGroupModule,
                    calendar.DxCalendarModule,
                    chart.DxChartModule,
                    checkBox.DxCheckBoxModule,
                    circularGauge.DxCircularGaugeModule,
                    colorBox.DxColorBoxModule,
                    contextMenu.DxContextMenuModule,
                    dataGrid.DxDataGridModule,
                    dateBox.DxDateBoxModule,
                    deferRendering.DxDeferRenderingModule,
                    diagram.DxDiagramModule,
                    draggable.DxDraggableModule,
                    drawer.DxDrawerModule,
                    dropDownBox.DxDropDownBoxModule,
                    dropDownButton.DxDropDownButtonModule,
                    fileManager.DxFileManagerModule,
                    fileUploader.DxFileUploaderModule,
                    filterBuilder.DxFilterBuilderModule,
                    form.DxFormModule,
                    funnel.DxFunnelModule,
                    gallery.DxGalleryModule,
                    gantt.DxGanttModule,
                    htmlEditor.DxHtmlEditorModule,
                    linearGauge.DxLinearGaugeModule,
                    list.DxListModule,
                    loadIndicator.DxLoadIndicatorModule,
                    loadPanel.DxLoadPanelModule,
                    lookup.DxLookupModule,
                    map.DxMapModule,
                    menu.DxMenuModule,
                    multiView.DxMultiViewModule,
                    navBar.DxNavBarModule,
                    numberBox.DxNumberBoxModule,
                    pieChart.DxPieChartModule,
                    pivotGrid.DxPivotGridModule,
                    pivotGridFieldChooser.DxPivotGridFieldChooserModule,
                    polarChart.DxPolarChartModule,
                    popover.DxPopoverModule,
                    popup.DxPopupModule,
                    progressBar.DxProgressBarModule,
                    radioGroup.DxRadioGroupModule,
                    rangeSelector.DxRangeSelectorModule,
                    rangeSlider.DxRangeSliderModule,
                    recurrenceEditor.DxRecurrenceEditorModule,
                    resizable.DxResizableModule,
                    responsiveBox.DxResponsiveBoxModule,
                    sankey.DxSankeyModule,
                    scheduler.DxSchedulerModule,
                    scrollView.DxScrollViewModule,
                    selectBox.DxSelectBoxModule,
                    slideOut.DxSlideOutModule,
                    slideOutView.DxSlideOutViewModule,
                    slider.DxSliderModule,
                    sortable.DxSortableModule,
                    sparkline.DxSparklineModule,
                    speedDialAction.DxSpeedDialActionModule,
                    _switch.DxSwitchModule,
                    tabPanel.DxTabPanelModule,
                    tabs.DxTabsModule,
                    tagBox.DxTagBoxModule,
                    textArea.DxTextAreaModule,
                    textBox.DxTextBoxModule,
                    tileView.DxTileViewModule,
                    toast.DxToastModule,
                    toolbar.DxToolbarModule,
                    tooltip.DxTooltipModule,
                    treeList.DxTreeListModule,
                    treeMap.DxTreeMapModule,
                    treeView.DxTreeViewModule,
                    validationGroup.DxValidationGroupModule,
                    validationSummary.DxValidationSummaryModule,
                    validator.DxValidatorModule,
                    vectorMap.DxVectorMapModule,
                    core.DxTemplateModule
                ],
                exports: [
                    accordion.DxAccordionModule,
                    actionSheet.DxActionSheetModule,
                    autocomplete.DxAutocompleteModule,
                    barGauge.DxBarGaugeModule,
                    box.DxBoxModule,
                    bullet.DxBulletModule,
                    button.DxButtonModule,
                    buttonGroup.DxButtonGroupModule,
                    calendar.DxCalendarModule,
                    chart.DxChartModule,
                    checkBox.DxCheckBoxModule,
                    circularGauge.DxCircularGaugeModule,
                    colorBox.DxColorBoxModule,
                    contextMenu.DxContextMenuModule,
                    dataGrid.DxDataGridModule,
                    dateBox.DxDateBoxModule,
                    deferRendering.DxDeferRenderingModule,
                    diagram.DxDiagramModule,
                    draggable.DxDraggableModule,
                    drawer.DxDrawerModule,
                    dropDownBox.DxDropDownBoxModule,
                    dropDownButton.DxDropDownButtonModule,
                    fileManager.DxFileManagerModule,
                    fileUploader.DxFileUploaderModule,
                    filterBuilder.DxFilterBuilderModule,
                    form.DxFormModule,
                    funnel.DxFunnelModule,
                    gallery.DxGalleryModule,
                    gantt.DxGanttModule,
                    htmlEditor.DxHtmlEditorModule,
                    linearGauge.DxLinearGaugeModule,
                    list.DxListModule,
                    loadIndicator.DxLoadIndicatorModule,
                    loadPanel.DxLoadPanelModule,
                    lookup.DxLookupModule,
                    map.DxMapModule,
                    menu.DxMenuModule,
                    multiView.DxMultiViewModule,
                    navBar.DxNavBarModule,
                    numberBox.DxNumberBoxModule,
                    pieChart.DxPieChartModule,
                    pivotGrid.DxPivotGridModule,
                    pivotGridFieldChooser.DxPivotGridFieldChooserModule,
                    polarChart.DxPolarChartModule,
                    popover.DxPopoverModule,
                    popup.DxPopupModule,
                    progressBar.DxProgressBarModule,
                    radioGroup.DxRadioGroupModule,
                    rangeSelector.DxRangeSelectorModule,
                    rangeSlider.DxRangeSliderModule,
                    recurrenceEditor.DxRecurrenceEditorModule,
                    resizable.DxResizableModule,
                    responsiveBox.DxResponsiveBoxModule,
                    sankey.DxSankeyModule,
                    scheduler.DxSchedulerModule,
                    scrollView.DxScrollViewModule,
                    selectBox.DxSelectBoxModule,
                    slideOut.DxSlideOutModule,
                    slideOutView.DxSlideOutViewModule,
                    slider.DxSliderModule,
                    sortable.DxSortableModule,
                    sparkline.DxSparklineModule,
                    speedDialAction.DxSpeedDialActionModule,
                    _switch.DxSwitchModule,
                    tabPanel.DxTabPanelModule,
                    tabs.DxTabsModule,
                    tagBox.DxTagBoxModule,
                    textArea.DxTextAreaModule,
                    textBox.DxTextBoxModule,
                    tileView.DxTileViewModule,
                    toast.DxToastModule,
                    toolbar.DxToolbarModule,
                    tooltip.DxTooltipModule,
                    treeList.DxTreeListModule,
                    treeMap.DxTreeMapModule,
                    treeView.DxTreeViewModule,
                    validationGroup.DxValidationGroupModule,
                    validationSummary.DxValidationSummaryModule,
                    validator.DxValidatorModule,
                    vectorMap.DxVectorMapModule,
                    core.DxTemplateModule
                ]
            })
        ], DevExtremeModule);
        return DevExtremeModule;
    }());

    Object.keys(core).forEach(function (k) {
        if (k !== 'default') Object.defineProperty(exports, k, {
            enumerable: true,
            get: function () {
                return core[k];
            }
        });
    });
    Object.defineProperty(exports, 'DxAccordionComponent', {
        enumerable: true,
        get: function () {
            return accordion.DxAccordionComponent;
        }
    });
    Object.defineProperty(exports, 'DxAccordionModule', {
        enumerable: true,
        get: function () {
            return accordion.DxAccordionModule;
        }
    });
    Object.defineProperty(exports, 'DxActionSheetComponent', {
        enumerable: true,
        get: function () {
            return actionSheet.DxActionSheetComponent;
        }
    });
    Object.defineProperty(exports, 'DxActionSheetModule', {
        enumerable: true,
        get: function () {
            return actionSheet.DxActionSheetModule;
        }
    });
    Object.defineProperty(exports, 'DxAutocompleteComponent', {
        enumerable: true,
        get: function () {
            return autocomplete.DxAutocompleteComponent;
        }
    });
    Object.defineProperty(exports, 'DxAutocompleteModule', {
        enumerable: true,
        get: function () {
            return autocomplete.DxAutocompleteModule;
        }
    });
    Object.defineProperty(exports, 'DxBarGaugeComponent', {
        enumerable: true,
        get: function () {
            return barGauge.DxBarGaugeComponent;
        }
    });
    Object.defineProperty(exports, 'DxBarGaugeModule', {
        enumerable: true,
        get: function () {
            return barGauge.DxBarGaugeModule;
        }
    });
    Object.defineProperty(exports, 'DxBoxComponent', {
        enumerable: true,
        get: function () {
            return box.DxBoxComponent;
        }
    });
    Object.defineProperty(exports, 'DxBoxModule', {
        enumerable: true,
        get: function () {
            return box.DxBoxModule;
        }
    });
    Object.defineProperty(exports, 'DxBulletComponent', {
        enumerable: true,
        get: function () {
            return bullet.DxBulletComponent;
        }
    });
    Object.defineProperty(exports, 'DxBulletModule', {
        enumerable: true,
        get: function () {
            return bullet.DxBulletModule;
        }
    });
    Object.defineProperty(exports, 'DxButtonComponent', {
        enumerable: true,
        get: function () {
            return button.DxButtonComponent;
        }
    });
    Object.defineProperty(exports, 'DxButtonModule', {
        enumerable: true,
        get: function () {
            return button.DxButtonModule;
        }
    });
    Object.defineProperty(exports, 'DxButtonGroupComponent', {
        enumerable: true,
        get: function () {
            return buttonGroup.DxButtonGroupComponent;
        }
    });
    Object.defineProperty(exports, 'DxButtonGroupModule', {
        enumerable: true,
        get: function () {
            return buttonGroup.DxButtonGroupModule;
        }
    });
    Object.defineProperty(exports, 'DxCalendarComponent', {
        enumerable: true,
        get: function () {
            return calendar.DxCalendarComponent;
        }
    });
    Object.defineProperty(exports, 'DxCalendarModule', {
        enumerable: true,
        get: function () {
            return calendar.DxCalendarModule;
        }
    });
    Object.defineProperty(exports, 'DxChartComponent', {
        enumerable: true,
        get: function () {
            return chart.DxChartComponent;
        }
    });
    Object.defineProperty(exports, 'DxChartModule', {
        enumerable: true,
        get: function () {
            return chart.DxChartModule;
        }
    });
    Object.defineProperty(exports, 'DxCheckBoxComponent', {
        enumerable: true,
        get: function () {
            return checkBox.DxCheckBoxComponent;
        }
    });
    Object.defineProperty(exports, 'DxCheckBoxModule', {
        enumerable: true,
        get: function () {
            return checkBox.DxCheckBoxModule;
        }
    });
    Object.defineProperty(exports, 'DxCircularGaugeComponent', {
        enumerable: true,
        get: function () {
            return circularGauge.DxCircularGaugeComponent;
        }
    });
    Object.defineProperty(exports, 'DxCircularGaugeModule', {
        enumerable: true,
        get: function () {
            return circularGauge.DxCircularGaugeModule;
        }
    });
    Object.defineProperty(exports, 'DxColorBoxComponent', {
        enumerable: true,
        get: function () {
            return colorBox.DxColorBoxComponent;
        }
    });
    Object.defineProperty(exports, 'DxColorBoxModule', {
        enumerable: true,
        get: function () {
            return colorBox.DxColorBoxModule;
        }
    });
    Object.defineProperty(exports, 'DxContextMenuComponent', {
        enumerable: true,
        get: function () {
            return contextMenu.DxContextMenuComponent;
        }
    });
    Object.defineProperty(exports, 'DxContextMenuModule', {
        enumerable: true,
        get: function () {
            return contextMenu.DxContextMenuModule;
        }
    });
    Object.defineProperty(exports, 'DxDataGridComponent', {
        enumerable: true,
        get: function () {
            return dataGrid.DxDataGridComponent;
        }
    });
    Object.defineProperty(exports, 'DxDataGridModule', {
        enumerable: true,
        get: function () {
            return dataGrid.DxDataGridModule;
        }
    });
    Object.defineProperty(exports, 'DxDateBoxComponent', {
        enumerable: true,
        get: function () {
            return dateBox.DxDateBoxComponent;
        }
    });
    Object.defineProperty(exports, 'DxDateBoxModule', {
        enumerable: true,
        get: function () {
            return dateBox.DxDateBoxModule;
        }
    });
    Object.defineProperty(exports, 'DxDeferRenderingComponent', {
        enumerable: true,
        get: function () {
            return deferRendering.DxDeferRenderingComponent;
        }
    });
    Object.defineProperty(exports, 'DxDeferRenderingModule', {
        enumerable: true,
        get: function () {
            return deferRendering.DxDeferRenderingModule;
        }
    });
    Object.defineProperty(exports, 'DxDiagramComponent', {
        enumerable: true,
        get: function () {
            return diagram.DxDiagramComponent;
        }
    });
    Object.defineProperty(exports, 'DxDiagramModule', {
        enumerable: true,
        get: function () {
            return diagram.DxDiagramModule;
        }
    });
    Object.defineProperty(exports, 'DxDraggableComponent', {
        enumerable: true,
        get: function () {
            return draggable.DxDraggableComponent;
        }
    });
    Object.defineProperty(exports, 'DxDraggableModule', {
        enumerable: true,
        get: function () {
            return draggable.DxDraggableModule;
        }
    });
    Object.defineProperty(exports, 'DxDrawerComponent', {
        enumerable: true,
        get: function () {
            return drawer.DxDrawerComponent;
        }
    });
    Object.defineProperty(exports, 'DxDrawerModule', {
        enumerable: true,
        get: function () {
            return drawer.DxDrawerModule;
        }
    });
    Object.defineProperty(exports, 'DxDropDownBoxComponent', {
        enumerable: true,
        get: function () {
            return dropDownBox.DxDropDownBoxComponent;
        }
    });
    Object.defineProperty(exports, 'DxDropDownBoxModule', {
        enumerable: true,
        get: function () {
            return dropDownBox.DxDropDownBoxModule;
        }
    });
    Object.defineProperty(exports, 'DxDropDownButtonComponent', {
        enumerable: true,
        get: function () {
            return dropDownButton.DxDropDownButtonComponent;
        }
    });
    Object.defineProperty(exports, 'DxDropDownButtonModule', {
        enumerable: true,
        get: function () {
            return dropDownButton.DxDropDownButtonModule;
        }
    });
    Object.defineProperty(exports, 'DxFileManagerComponent', {
        enumerable: true,
        get: function () {
            return fileManager.DxFileManagerComponent;
        }
    });
    Object.defineProperty(exports, 'DxFileManagerModule', {
        enumerable: true,
        get: function () {
            return fileManager.DxFileManagerModule;
        }
    });
    Object.defineProperty(exports, 'DxFileUploaderComponent', {
        enumerable: true,
        get: function () {
            return fileUploader.DxFileUploaderComponent;
        }
    });
    Object.defineProperty(exports, 'DxFileUploaderModule', {
        enumerable: true,
        get: function () {
            return fileUploader.DxFileUploaderModule;
        }
    });
    Object.defineProperty(exports, 'DxFilterBuilderComponent', {
        enumerable: true,
        get: function () {
            return filterBuilder.DxFilterBuilderComponent;
        }
    });
    Object.defineProperty(exports, 'DxFilterBuilderModule', {
        enumerable: true,
        get: function () {
            return filterBuilder.DxFilterBuilderModule;
        }
    });
    Object.defineProperty(exports, 'DxFormComponent', {
        enumerable: true,
        get: function () {
            return form.DxFormComponent;
        }
    });
    Object.defineProperty(exports, 'DxFormModule', {
        enumerable: true,
        get: function () {
            return form.DxFormModule;
        }
    });
    Object.defineProperty(exports, 'DxFunnelComponent', {
        enumerable: true,
        get: function () {
            return funnel.DxFunnelComponent;
        }
    });
    Object.defineProperty(exports, 'DxFunnelModule', {
        enumerable: true,
        get: function () {
            return funnel.DxFunnelModule;
        }
    });
    Object.defineProperty(exports, 'DxGalleryComponent', {
        enumerable: true,
        get: function () {
            return gallery.DxGalleryComponent;
        }
    });
    Object.defineProperty(exports, 'DxGalleryModule', {
        enumerable: true,
        get: function () {
            return gallery.DxGalleryModule;
        }
    });
    Object.defineProperty(exports, 'DxGanttComponent', {
        enumerable: true,
        get: function () {
            return gantt.DxGanttComponent;
        }
    });
    Object.defineProperty(exports, 'DxGanttModule', {
        enumerable: true,
        get: function () {
            return gantt.DxGanttModule;
        }
    });
    Object.defineProperty(exports, 'DxHtmlEditorComponent', {
        enumerable: true,
        get: function () {
            return htmlEditor.DxHtmlEditorComponent;
        }
    });
    Object.defineProperty(exports, 'DxHtmlEditorModule', {
        enumerable: true,
        get: function () {
            return htmlEditor.DxHtmlEditorModule;
        }
    });
    Object.defineProperty(exports, 'DxLinearGaugeComponent', {
        enumerable: true,
        get: function () {
            return linearGauge.DxLinearGaugeComponent;
        }
    });
    Object.defineProperty(exports, 'DxLinearGaugeModule', {
        enumerable: true,
        get: function () {
            return linearGauge.DxLinearGaugeModule;
        }
    });
    Object.defineProperty(exports, 'DxListComponent', {
        enumerable: true,
        get: function () {
            return list.DxListComponent;
        }
    });
    Object.defineProperty(exports, 'DxListModule', {
        enumerable: true,
        get: function () {
            return list.DxListModule;
        }
    });
    Object.defineProperty(exports, 'DxLoadIndicatorComponent', {
        enumerable: true,
        get: function () {
            return loadIndicator.DxLoadIndicatorComponent;
        }
    });
    Object.defineProperty(exports, 'DxLoadIndicatorModule', {
        enumerable: true,
        get: function () {
            return loadIndicator.DxLoadIndicatorModule;
        }
    });
    Object.defineProperty(exports, 'DxLoadPanelComponent', {
        enumerable: true,
        get: function () {
            return loadPanel.DxLoadPanelComponent;
        }
    });
    Object.defineProperty(exports, 'DxLoadPanelModule', {
        enumerable: true,
        get: function () {
            return loadPanel.DxLoadPanelModule;
        }
    });
    Object.defineProperty(exports, 'DxLookupComponent', {
        enumerable: true,
        get: function () {
            return lookup.DxLookupComponent;
        }
    });
    Object.defineProperty(exports, 'DxLookupModule', {
        enumerable: true,
        get: function () {
            return lookup.DxLookupModule;
        }
    });
    Object.defineProperty(exports, 'DxMapComponent', {
        enumerable: true,
        get: function () {
            return map.DxMapComponent;
        }
    });
    Object.defineProperty(exports, 'DxMapModule', {
        enumerable: true,
        get: function () {
            return map.DxMapModule;
        }
    });
    Object.defineProperty(exports, 'DxMenuComponent', {
        enumerable: true,
        get: function () {
            return menu.DxMenuComponent;
        }
    });
    Object.defineProperty(exports, 'DxMenuModule', {
        enumerable: true,
        get: function () {
            return menu.DxMenuModule;
        }
    });
    Object.defineProperty(exports, 'DxMultiViewComponent', {
        enumerable: true,
        get: function () {
            return multiView.DxMultiViewComponent;
        }
    });
    Object.defineProperty(exports, 'DxMultiViewModule', {
        enumerable: true,
        get: function () {
            return multiView.DxMultiViewModule;
        }
    });
    Object.defineProperty(exports, 'DxNavBarComponent', {
        enumerable: true,
        get: function () {
            return navBar.DxNavBarComponent;
        }
    });
    Object.defineProperty(exports, 'DxNavBarModule', {
        enumerable: true,
        get: function () {
            return navBar.DxNavBarModule;
        }
    });
    Object.defineProperty(exports, 'DxNumberBoxComponent', {
        enumerable: true,
        get: function () {
            return numberBox.DxNumberBoxComponent;
        }
    });
    Object.defineProperty(exports, 'DxNumberBoxModule', {
        enumerable: true,
        get: function () {
            return numberBox.DxNumberBoxModule;
        }
    });
    Object.defineProperty(exports, 'DxPieChartComponent', {
        enumerable: true,
        get: function () {
            return pieChart.DxPieChartComponent;
        }
    });
    Object.defineProperty(exports, 'DxPieChartModule', {
        enumerable: true,
        get: function () {
            return pieChart.DxPieChartModule;
        }
    });
    Object.defineProperty(exports, 'DxPivotGridComponent', {
        enumerable: true,
        get: function () {
            return pivotGrid.DxPivotGridComponent;
        }
    });
    Object.defineProperty(exports, 'DxPivotGridModule', {
        enumerable: true,
        get: function () {
            return pivotGrid.DxPivotGridModule;
        }
    });
    Object.defineProperty(exports, 'DxPivotGridFieldChooserComponent', {
        enumerable: true,
        get: function () {
            return pivotGridFieldChooser.DxPivotGridFieldChooserComponent;
        }
    });
    Object.defineProperty(exports, 'DxPivotGridFieldChooserModule', {
        enumerable: true,
        get: function () {
            return pivotGridFieldChooser.DxPivotGridFieldChooserModule;
        }
    });
    Object.defineProperty(exports, 'DxPolarChartComponent', {
        enumerable: true,
        get: function () {
            return polarChart.DxPolarChartComponent;
        }
    });
    Object.defineProperty(exports, 'DxPolarChartModule', {
        enumerable: true,
        get: function () {
            return polarChart.DxPolarChartModule;
        }
    });
    Object.defineProperty(exports, 'DxPopoverComponent', {
        enumerable: true,
        get: function () {
            return popover.DxPopoverComponent;
        }
    });
    Object.defineProperty(exports, 'DxPopoverModule', {
        enumerable: true,
        get: function () {
            return popover.DxPopoverModule;
        }
    });
    Object.defineProperty(exports, 'DxPopupComponent', {
        enumerable: true,
        get: function () {
            return popup.DxPopupComponent;
        }
    });
    Object.defineProperty(exports, 'DxPopupModule', {
        enumerable: true,
        get: function () {
            return popup.DxPopupModule;
        }
    });
    Object.defineProperty(exports, 'DxProgressBarComponent', {
        enumerable: true,
        get: function () {
            return progressBar.DxProgressBarComponent;
        }
    });
    Object.defineProperty(exports, 'DxProgressBarModule', {
        enumerable: true,
        get: function () {
            return progressBar.DxProgressBarModule;
        }
    });
    Object.defineProperty(exports, 'DxRadioGroupComponent', {
        enumerable: true,
        get: function () {
            return radioGroup.DxRadioGroupComponent;
        }
    });
    Object.defineProperty(exports, 'DxRadioGroupModule', {
        enumerable: true,
        get: function () {
            return radioGroup.DxRadioGroupModule;
        }
    });
    Object.defineProperty(exports, 'DxRangeSelectorComponent', {
        enumerable: true,
        get: function () {
            return rangeSelector.DxRangeSelectorComponent;
        }
    });
    Object.defineProperty(exports, 'DxRangeSelectorModule', {
        enumerable: true,
        get: function () {
            return rangeSelector.DxRangeSelectorModule;
        }
    });
    Object.defineProperty(exports, 'DxRangeSliderComponent', {
        enumerable: true,
        get: function () {
            return rangeSlider.DxRangeSliderComponent;
        }
    });
    Object.defineProperty(exports, 'DxRangeSliderModule', {
        enumerable: true,
        get: function () {
            return rangeSlider.DxRangeSliderModule;
        }
    });
    Object.defineProperty(exports, 'DxRecurrenceEditorComponent', {
        enumerable: true,
        get: function () {
            return recurrenceEditor.DxRecurrenceEditorComponent;
        }
    });
    Object.defineProperty(exports, 'DxRecurrenceEditorModule', {
        enumerable: true,
        get: function () {
            return recurrenceEditor.DxRecurrenceEditorModule;
        }
    });
    Object.defineProperty(exports, 'DxResizableComponent', {
        enumerable: true,
        get: function () {
            return resizable.DxResizableComponent;
        }
    });
    Object.defineProperty(exports, 'DxResizableModule', {
        enumerable: true,
        get: function () {
            return resizable.DxResizableModule;
        }
    });
    Object.defineProperty(exports, 'DxResponsiveBoxComponent', {
        enumerable: true,
        get: function () {
            return responsiveBox.DxResponsiveBoxComponent;
        }
    });
    Object.defineProperty(exports, 'DxResponsiveBoxModule', {
        enumerable: true,
        get: function () {
            return responsiveBox.DxResponsiveBoxModule;
        }
    });
    Object.defineProperty(exports, 'DxSankeyComponent', {
        enumerable: true,
        get: function () {
            return sankey.DxSankeyComponent;
        }
    });
    Object.defineProperty(exports, 'DxSankeyModule', {
        enumerable: true,
        get: function () {
            return sankey.DxSankeyModule;
        }
    });
    Object.defineProperty(exports, 'DxSchedulerComponent', {
        enumerable: true,
        get: function () {
            return scheduler.DxSchedulerComponent;
        }
    });
    Object.defineProperty(exports, 'DxSchedulerModule', {
        enumerable: true,
        get: function () {
            return scheduler.DxSchedulerModule;
        }
    });
    Object.defineProperty(exports, 'DxScrollViewComponent', {
        enumerable: true,
        get: function () {
            return scrollView.DxScrollViewComponent;
        }
    });
    Object.defineProperty(exports, 'DxScrollViewModule', {
        enumerable: true,
        get: function () {
            return scrollView.DxScrollViewModule;
        }
    });
    Object.defineProperty(exports, 'DxSelectBoxComponent', {
        enumerable: true,
        get: function () {
            return selectBox.DxSelectBoxComponent;
        }
    });
    Object.defineProperty(exports, 'DxSelectBoxModule', {
        enumerable: true,
        get: function () {
            return selectBox.DxSelectBoxModule;
        }
    });
    Object.defineProperty(exports, 'DxSlideOutComponent', {
        enumerable: true,
        get: function () {
            return slideOut.DxSlideOutComponent;
        }
    });
    Object.defineProperty(exports, 'DxSlideOutModule', {
        enumerable: true,
        get: function () {
            return slideOut.DxSlideOutModule;
        }
    });
    Object.defineProperty(exports, 'DxSlideOutViewComponent', {
        enumerable: true,
        get: function () {
            return slideOutView.DxSlideOutViewComponent;
        }
    });
    Object.defineProperty(exports, 'DxSlideOutViewModule', {
        enumerable: true,
        get: function () {
            return slideOutView.DxSlideOutViewModule;
        }
    });
    Object.defineProperty(exports, 'DxSliderComponent', {
        enumerable: true,
        get: function () {
            return slider.DxSliderComponent;
        }
    });
    Object.defineProperty(exports, 'DxSliderModule', {
        enumerable: true,
        get: function () {
            return slider.DxSliderModule;
        }
    });
    Object.defineProperty(exports, 'DxSortableComponent', {
        enumerable: true,
        get: function () {
            return sortable.DxSortableComponent;
        }
    });
    Object.defineProperty(exports, 'DxSortableModule', {
        enumerable: true,
        get: function () {
            return sortable.DxSortableModule;
        }
    });
    Object.defineProperty(exports, 'DxSparklineComponent', {
        enumerable: true,
        get: function () {
            return sparkline.DxSparklineComponent;
        }
    });
    Object.defineProperty(exports, 'DxSparklineModule', {
        enumerable: true,
        get: function () {
            return sparkline.DxSparklineModule;
        }
    });
    Object.defineProperty(exports, 'DxSpeedDialActionComponent', {
        enumerable: true,
        get: function () {
            return speedDialAction.DxSpeedDialActionComponent;
        }
    });
    Object.defineProperty(exports, 'DxSpeedDialActionModule', {
        enumerable: true,
        get: function () {
            return speedDialAction.DxSpeedDialActionModule;
        }
    });
    Object.defineProperty(exports, 'DxSwitchComponent', {
        enumerable: true,
        get: function () {
            return _switch.DxSwitchComponent;
        }
    });
    Object.defineProperty(exports, 'DxSwitchModule', {
        enumerable: true,
        get: function () {
            return _switch.DxSwitchModule;
        }
    });
    Object.defineProperty(exports, 'DxTabPanelComponent', {
        enumerable: true,
        get: function () {
            return tabPanel.DxTabPanelComponent;
        }
    });
    Object.defineProperty(exports, 'DxTabPanelModule', {
        enumerable: true,
        get: function () {
            return tabPanel.DxTabPanelModule;
        }
    });
    Object.defineProperty(exports, 'DxTabsComponent', {
        enumerable: true,
        get: function () {
            return tabs.DxTabsComponent;
        }
    });
    Object.defineProperty(exports, 'DxTabsModule', {
        enumerable: true,
        get: function () {
            return tabs.DxTabsModule;
        }
    });
    Object.defineProperty(exports, 'DxTagBoxComponent', {
        enumerable: true,
        get: function () {
            return tagBox.DxTagBoxComponent;
        }
    });
    Object.defineProperty(exports, 'DxTagBoxModule', {
        enumerable: true,
        get: function () {
            return tagBox.DxTagBoxModule;
        }
    });
    Object.defineProperty(exports, 'DxTextAreaComponent', {
        enumerable: true,
        get: function () {
            return textArea.DxTextAreaComponent;
        }
    });
    Object.defineProperty(exports, 'DxTextAreaModule', {
        enumerable: true,
        get: function () {
            return textArea.DxTextAreaModule;
        }
    });
    Object.defineProperty(exports, 'DxTextBoxComponent', {
        enumerable: true,
        get: function () {
            return textBox.DxTextBoxComponent;
        }
    });
    Object.defineProperty(exports, 'DxTextBoxModule', {
        enumerable: true,
        get: function () {
            return textBox.DxTextBoxModule;
        }
    });
    Object.defineProperty(exports, 'DxTileViewComponent', {
        enumerable: true,
        get: function () {
            return tileView.DxTileViewComponent;
        }
    });
    Object.defineProperty(exports, 'DxTileViewModule', {
        enumerable: true,
        get: function () {
            return tileView.DxTileViewModule;
        }
    });
    Object.defineProperty(exports, 'DxToastComponent', {
        enumerable: true,
        get: function () {
            return toast.DxToastComponent;
        }
    });
    Object.defineProperty(exports, 'DxToastModule', {
        enumerable: true,
        get: function () {
            return toast.DxToastModule;
        }
    });
    Object.defineProperty(exports, 'DxToolbarComponent', {
        enumerable: true,
        get: function () {
            return toolbar.DxToolbarComponent;
        }
    });
    Object.defineProperty(exports, 'DxToolbarModule', {
        enumerable: true,
        get: function () {
            return toolbar.DxToolbarModule;
        }
    });
    Object.defineProperty(exports, 'DxTooltipComponent', {
        enumerable: true,
        get: function () {
            return tooltip.DxTooltipComponent;
        }
    });
    Object.defineProperty(exports, 'DxTooltipModule', {
        enumerable: true,
        get: function () {
            return tooltip.DxTooltipModule;
        }
    });
    Object.defineProperty(exports, 'DxTreeListComponent', {
        enumerable: true,
        get: function () {
            return treeList.DxTreeListComponent;
        }
    });
    Object.defineProperty(exports, 'DxTreeListModule', {
        enumerable: true,
        get: function () {
            return treeList.DxTreeListModule;
        }
    });
    Object.defineProperty(exports, 'DxTreeMapComponent', {
        enumerable: true,
        get: function () {
            return treeMap.DxTreeMapComponent;
        }
    });
    Object.defineProperty(exports, 'DxTreeMapModule', {
        enumerable: true,
        get: function () {
            return treeMap.DxTreeMapModule;
        }
    });
    Object.defineProperty(exports, 'DxTreeViewComponent', {
        enumerable: true,
        get: function () {
            return treeView.DxTreeViewComponent;
        }
    });
    Object.defineProperty(exports, 'DxTreeViewModule', {
        enumerable: true,
        get: function () {
            return treeView.DxTreeViewModule;
        }
    });
    Object.defineProperty(exports, 'DxValidationGroupComponent', {
        enumerable: true,
        get: function () {
            return validationGroup.DxValidationGroupComponent;
        }
    });
    Object.defineProperty(exports, 'DxValidationGroupModule', {
        enumerable: true,
        get: function () {
            return validationGroup.DxValidationGroupModule;
        }
    });
    Object.defineProperty(exports, 'DxValidationSummaryComponent', {
        enumerable: true,
        get: function () {
            return validationSummary.DxValidationSummaryComponent;
        }
    });
    Object.defineProperty(exports, 'DxValidationSummaryModule', {
        enumerable: true,
        get: function () {
            return validationSummary.DxValidationSummaryModule;
        }
    });
    Object.defineProperty(exports, 'DxValidatorComponent', {
        enumerable: true,
        get: function () {
            return validator.DxValidatorComponent;
        }
    });
    Object.defineProperty(exports, 'DxValidatorModule', {
        enumerable: true,
        get: function () {
            return validator.DxValidatorModule;
        }
    });
    Object.defineProperty(exports, 'DxVectorMapComponent', {
        enumerable: true,
        get: function () {
            return vectorMap.DxVectorMapComponent;
        }
    });
    Object.defineProperty(exports, 'DxVectorMapModule', {
        enumerable: true,
        get: function () {
            return vectorMap.DxVectorMapModule;
        }
    });
    exports.DevExtremeModule = DevExtremeModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
